src/python/pants/rules/core/generate_pants_ini.py

```diff
@@ -16,36 +16,50 @@
 from pants.engine.goal import Goal, GoalSubsystem
 from pants.engine.rules import goal_rule
 from pants.engine.selectors import Get
-from pants.version import VERSION as pants_version
+from pants.version import VERSION
 
 
 class GeneratePantsIniOptions(GoalSubsystem):
   """Generate pants.ini with sensible defaults."""
   name = 'generate-pants-ini'
 
+  @classmethod
+  def register_options(cls, register) -> None:
+    super().register_options(register)
+    register('--v2-only', type=bool, fingerprint=True, default=False,
+             help='Support only v2 rules. If unspecified, this repo will also run v1 tasks.')
+
 
 class GeneratePantsIni(Goal):
   subsystem_cls = GeneratePantsIniOptions
 
 
 @goal_rule
-async def generate_pants_ini(console: Console, workspace: Workspace) -> GeneratePantsIni:
+async def generate_pants_ini(console: Console, workspace: Workspace,
+                             options: GeneratePantsIniOptions) -> GeneratePantsIni:
   pants_ini_content = dedent(f"""\
     [GLOBAL]
-    pants_version: {pants_version}
+    pants_version: {VERSION}
+    v1: {'False' if options.values.v2_only else 'True'}
+    v2: True
+    v2_ui: {'True' if options.values.v2_only else 'False'}
+    backend_packages: []
+    plugins: []
+    backend_packages2: []
+    plugins2: []
     """)
 
   preexisting_snapshot = await Get[Snapshot](PathGlobs(include=('pants.ini',)))
   if preexisting_snapshot.files:
     console.print_stderr(
-      "./pants.ini already exists. This goal is only meant to be run the first time you run Pants "
-      "in a project.\n\nTo update config values, please directly modify the file."
+      "./pants.ini already exists. This goal is only meant to be run to set up Pants "
+      "in a repo for the first time.\n\nTo update config values, please directly modify the file."
     )
     return GeneratePantsIni(exit_code=1)
 
   console.print_stdout(dedent(f"""\
-    Adding sensible defaults to ./pants.ini:
-      * Pinning `pants_version` to `{pants_version}`.
+      * Adding sensible defaults to ./pants.ini.
+      * Pinning `pants_version` to `{VERSION}`.
```

Comment:
The idea of this original setup is to print to the user what each option is being set with some explanation for what it does.

For example,

```
Adding sensible defaults to pants.ini:
  * Pinning `pants_version` to `{VERSION}`.
  * Disabling the V1 backend and enabling the V2 backend.
  * Setting up empty `backend_packages` and `plugins` sections so that you may easily add values. See https://..... for information on what values to put.
```

This is the first command ever run by new Pants users, so we want to find a balance where we're helpful and transparent, but also not overwhelming.

Reply:
It just feels a bit redundant to recapitulate the contents we just generated, but I'll add something.